src/formState.ts

```diff
@@ -673,7 +673,7 @@ function newValueFieldState<T, K extends keyof T>(
         throw new Error(`${key} is currently readOnly`);
       }
 
-      if (opts.refreshing && this.dirty) {
+      if (opts.refreshing && this.dirty && value !== this.value) {
```

Comment:
This was the bug; if you had changed firstName to "bob", and then saved, the query cache would update and come in with "hey firstName is now bob", but we'd ignore it b/c we're dirty.

Instead we got ahead and set this "set" happen, since both our internal value and the new-from-server value are both "bob" now, which ends up setting `originalValue=bob` and then we're no longer dirty.